docs/standard/garbage-collection/large-object-heap.md

```diff
@@ -0,0 +1,313 @@
+ 
+![Figure 6: A GC Heap Alloc view](media/loh/perfview2.png)  
+Figure 6: A GC Heap Alloc view
+ 
+As you can see, this is a very simple test that just allocates large objects from its `Main` method.
+
+### A debugger
+
+If all you have is a memory dump and you need to look at what objects are actually on the LOH, you can use the [SoS debugger extension](http://msdn2.microsoft.com/ms404370.aspx) provided by .NET. 
+
+> [!NOTE]
+> The debugging commands mentioned in this section are applicable to the [Windows Debuggers](http://www.microsoft.com/whdc/devtools/debugging/default.mspx).
+
+The following shows sample output from analyzing the LOH:
+
+```
+0:003> .loadby sos mscorwks
+0:003> !eeheap -gc
+Number of GC Heaps: 1
+generation 0 starts at 0x013e35ec
+sdgeneration 1 starts at 0x013e1b6c
+generation 2 starts at 0x013e1000
+ephemeral segment allocation context: none
+segment   begin allocated     size
+0018f2d0 790d5588 790f4b38 0x0001f5b0(128432)
+013e0000 013e1000 013e35f8 0x000025f8(9720)
+Large object heap starts at 0x023e1000
+segment   begin allocated     size
+023e0000 023e1000 033db630 0x00ffa630(16754224)
+033e0000 033e1000 043cdf98 0x00fecf98(16699288)
+043e0000 043e1000 05368b58 0x00f87b58(16284504)
+Total Size 0x2f90cc8(49876168)
+------------------------------
+GC Heap Size 0x2f90cc8(49876168)
+0:003> !dumpheap -stat 023e1000 033db630
+total 133 objects
+Statistics:
+MT   Count   TotalSize Class Name
+001521d0       66     2081792     Free
+7912273c       63     6663696 System.Byte[]
+7912254c       4     8008736 System.Object[]
+Total 133 objects
+```
+
+The LOH heap size is (16,754,224 + 16,699,288 + 16,284,504) = 49,738,016 bytes. Between addresses 023e1000 and 033db630, 8,008,736 bytes are occupied by an array of <xref:System.Object?displayProperty=fullName> objects, 6,663,696 bytes are occupied by an array of <xref:System.Byte?displayProperty=nameWithType>  objects, and 2,081,792 bytes are occupied by free space.
+
+Sometimes, the debugger shows that the total size of the LOH is less than 85,000 bytes. This happens because the runtime itself uses the LOH to allocate some objects that are smaller than a large object.
+
+Because the LOH is not compacted, sometimes the LOH is thoought to be the source of fragmentation. Fragmentation means:
+
+- Fragmentation of the managed heap, which is indicated by the amount of free space between managed objects. In SoS, the `!dumpheap –type Free` command displays the amount of free space between managed objects.
+
+- Fragmentation of the virtual memory address space, which is the memory marked as `MEM_FREE`. You can get it by using various debugger commands in windbg. 
+
+   The following example shows fragmentation in the VM space:
```

Comment:
here, and a couple locations below, I would spell out "Virtual Memory", or introduce the abbreviation VM to be Virutal Memory on first use. (VM could be Virtual Machine).